source/cl/test/UnitCL/source/cl_khr_command_buffer/clCommandNDRangeKernelKHR.cpp

```diff
@@ -72,6 +72,55 @@ TEST_F(CommandNDRangeKernelTest, EmptyKernel) {
   ASSERT_SUCCESS(clReleaseProgram(program));
 }
 
+// Tests the whether we can enqueue an execute a command buffer containing a
+// kernel enqueue created via clCommandNDRangeKernelKHR with global size of 0
+// or a global size of nullptr
+TEST_F(CommandNDRangeKernelTest, GlobalZero) {
+  // Requires a compiler to compile the kernel.
+  if (!getDeviceCompilerAvailable()) {
+    GTEST_SKIP();
+  }
+  // Set up the kernel.
+  const char *code = R"OpenCLC(
+  __kernel void empty_kernel() { __builtin_trap(); }
```

Comment:
I know I was the one who suggested this, but I forgot that UnitCL is meant to be portable across OpenCL implementations and `__builtin_trap` is not required to be supported. We could put this in an `#ifdef __clang__` or such?